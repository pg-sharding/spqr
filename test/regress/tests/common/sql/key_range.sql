CREATE DISTRIBUTION ds1 COLUMN TYPES integer;

-- create key range works
CREATE KEY RANGE krid1 FROM 0 ROUTE TO sh1 FOR DISTRIBUTION ds1;
CREATE KEY RANGE krid2 FROM 11 ROUTE TO sh2 FOR DISTRIBUTION ds1;

-- intersecting key range
CREATE KEY RANGE krid3 FROM 11 ROUTE TO sh1 FOR DISTRIBUTION ds1;
