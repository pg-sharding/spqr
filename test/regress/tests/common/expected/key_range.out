CREATE DISTRIBUTION ds1 COLUMN TYPES integer;
    add distribution    
------------------------
 distribution id -> ds1
(1 row)

-- create key range works 
CREATE KEY RANGE krid2 FROM 11 ROUTE TO sh2 FOR DISTRIBUTION ds1;
 add key range 
---------------
 bound -> 11
(1 row)

CREATE KEY RANGE krid1 FROM 0 ROUTE TO sh1 FOR DISTRIBUTION ds1;
 add key range 
---------------
 bound -> 0
(1 row)

-- intersecting key range
CREATE KEY RANGE krid3 FROM 50 ROUTE TO sh1 FOR DISTRIBUTION ds1;
ERROR:  key range krid3 intersects with key range krid2 in QDB.
